core/src/main/java/nl/nn/adapterframework/pgp/PGPAction.java

```diff
@@ -0,0 +1,91 @@
+package nl.nn.adapterframework.pgp;
+
+import name.neuhalfen.projects.crypto.bouncycastle.openpgp.keys.callbacks.KeyringConfigCallback;
+import name.neuhalfen.projects.crypto.bouncycastle.openpgp.keys.callbacks.KeyringConfigCallbacks;
+import name.neuhalfen.projects.crypto.bouncycastle.openpgp.keys.keyrings.InMemoryKeyring;
+import name.neuhalfen.projects.crypto.bouncycastle.openpgp.keys.keyrings.KeyringConfigs;
+import nl.nn.adapterframework.configuration.ConfigurationException;
+import org.apache.commons.io.IOUtils;
+import org.bouncycastle.openpgp.PGPException;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+/**
+ * This is an abstraction of general pgp actions
+ * such as encryption, verification, etc.
+ * to be used for {@link nl.nn.adapterframework.pipes.PGPPipe}
+ *
+ * @author Murat Kaan Meral
+ */
+public abstract class PGPAction {
+
+	protected InMemoryKeyring keyringConfig;
+	private String[] publicKey;
```

Comment:
shouldn't the name be the plural publicKeys?

Reply:
Yeah it should be, also the one in pgp pipe also should be publickeys